src/averageAges.js

```diff
@@ -1,61 +1,69 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => century
+    ? Math.ceil(person.died / 100) === century
+      && person.sex === 'm'
+    : person.sex === 'm');
+
+  const menSum = men.reduce(
+    (sum, person) => sum + (person.died - person.born), 0);
+  const average = menSum / men.length;
+
+  return average;
```

Comment:
This logic  for calculating average age is common for several functions, it would be nice to create a separate function to calculate this value to not repeat